### PR TITLE
# Release 4.2.0

### PR DESCRIPTION
**Pull request by golluroSICKAG**
# Version 4.2.0

## New features
- Provide version of module via 'OnNewStatusModuleVersion'
- Function 'getParameters' to provide PersistentData parameters
- Check if features of module can be used on device and provide this via 'OnNewStatusModuleIsActive' event / 'getStatusModuleActive' function

## Improvements
- New UI design available (e.g. selectable via CSK_Module_PersistentData v4.1.0 or higher), see 'OnNewStatusCSKStyle'
- 'loadParameters' returns its success
- 'sendParameters' can control if sent data should be saved directly by CSK_Module_PersistentData
- Changed log level of some messages from 'info' to 'fine'
- Added UI icon and browser tab information

## Following tasks were checked

- [x] Module is coded and structured according to the "Developing guideline for modules" described within the CSK documentation
- [x] All functions/events/parameters are documented within the manifest documentation
- [x] The manifest description of the main CROWN includes main information about the purpose of the module and how to use it in general
- [x] API docu based on the manifest was created and stored within the "docu"-folder of the repository
- [x] Internal LUA code documentation exists for variables and non served functions
- [x] All relevant infos are logged via the SharedLogger 'ModuleLogger'
- [x] Module supports persistent data feature based on 'CSK_Module_PersistentData'
- [x] Module supports user management based on 'CSK_Module_UserManagement'
- [x] No open "ToDos" within the code or at least clearly explained comments why they exist...
- [x] "Version" key in app manifest was updated following semantic versioning (and use '0.x.y' for test / experimental modules which are not yet ready to be officially released)
- [x] Meaningful IDs used for UI elements
- [x] Module was tested on an AppSpace device (at least on SICK AppEngine) with no error message
- [x] README.md is up to date (incl. info of device + firmware the module was tested with)
- [x] CHANGELOG.md is up to date

### CODE REVIEW

File: CSK_1stModule_Logger/pages/pages/CSK_Module_Logger/CSK_Module_Logger.html [hunk 1]
```diff
@@ -1,347 +1,433 @@
...
+                                                            update-on-resume>
+                                                        </crown-edpws-binding>
+                                                        <crown-binding event="change" name="CSK_Logger/setFilePath"
+                                                            path="param/args/path" auto-commit>
+                                                        </crown-binding>
+                                                    </davinci-text-field>
+                                                </layout-column>
+                                                <layout-column id="ColumnLayout13" style="align-items: stretch">
+                                                    <davinci-text-field id="TF_Filename" type="text"
+                                                        title="Name of the log file to store log mesasgs.">
```
Comment (kogliinSICKAG) [inline CSK_1stModule_Logger/pages/pages/CSK_Module_Logger/CSK_Module_Logger.html]: wrong spelling: messages

File: CSK_1stModule_Logger/pages/pages/CSK_Module_Logger/CSK_Module_Logger.html [hunk 2]
```diff
@@ -1,347 +1,433 @@
...
+                                                </layout-column>
+                                                <layout-column id="ColumnLayout33" style="align-items: stretch">
+                                                </layout-column>
+                                                <layout-column id="ColumnLayout32" style="align-items: stretch">
+                                                    <davinci-collapse id="Collapse_FileSinkInfo" collapsed-title="Info"
+                                                        expanded-title="Hide" style="align-self: center">
+                                                        <davinci-callout id="Callout3" type="info">
+                                                            <span>
+		        											If using the 'file sink' feature, a SD card should be used to reduce write accesses on the internal flash storage of the device.
+		        											</span>
```
Comment (kogliinSICKAG) [inline CSK_1stModule_Logger/pages/pages/CSK_Module_Logger/CSK_Module_Logger.html]: two lines with tabs as indentation

File: CSK_1stModule_Logger/pages/pages/CSK_Module_Logger/CSK_Module_Logger.html [hunk 3]
```diff
@@ -1,347 +1,433 @@
...
+                                            label="Log messages">
+                                            <crown-edpws-binding property="value" name="CSK_Logger/OnNewCompleteLogfile"
+                                                update-on-resume>
+                                            </crown-edpws-binding>
+                                        </davinci-text-area>
+                                        <layout-row id="RowLayout3" class="myMargin20px0_CSK_Module_Logger"
+                                            style="justify-content: flex-end">
+                                            <appspace-file-download-button id="FDB_DownloadFile" type="outline"
+                                                icon-position="append" downloaded-filename="file.tmp"
+                                                title="Download Logfile from device to the pc." icon="file/file_download">
```
Comment (kogliinSICKAG) [inline CSK_1stModule_Logger/pages/pages/CSK_Module_Logger/CSK_Module_Logger.html]: abbreviation for personal computer should be in capital letters: PC

File: CSK_1stModule_Logger/pages/pages/CSK_Module_Logger/CSK_Module_Logger.html [hunk 4]
```diff
@@ -1,347 +1,433 @@
...
+        <stacked-view id="SV_PageCalled">
+            <crown-binding property="value" name="CSK_Logger/pageCalled" path="result/data/empty" auto-update="0"
+                update-on-resume>
+            </crown-binding>
+            <crown-edpws-binding property="value" name="CSK_Logger/OnNewStatusCSKStyle" update-on-resume
+                converter="changeStyle">
+            </crown-edpws-binding>
+        </stacked-view>
+    </layout-column>
 </layout-row>
```
Comment (kogliinSICKAG) [inline CSK_1stModule_Logger/pages/pages/CSK_Module_Logger/CSK_Module_Logger.html:470]: I did not run the GUI yet but from the code I could see some major changes. Are these changes like taking out the display of the last log message made due to the redesign of GUI for a better user experience?

**Review by kogliinSICKAG — COMMENTED** [review]
Some minor changes.

File: CSK_1stModule_Logger/pages/pages/CSK_Module_Logger/CSK_Module_Logger.html (hunk 4 above)
Comment (golluroSICKAG) [reply CSK_1stModule_Logger/pages/pages/CSK_Module_Logger/CSK_Module_Logger.html:470]: Yes, exactly

**Comment by golluroSICKAG** [comment]
Thx for the review!

**Pull request merged**